### PR TITLE
Fyle 1v1hkjc : Fixed 'Please add a project ' alignment in Expenses

### PR DESCRIPTION
**Pull request by siddharth0-0**
image

image

### CODE REVIEW

**Comment by Dimple16** [comment]
Is this tested for the below cases:
- When projects are assigned 
- When projects are not assigned but it is mandatory 
- When billable is checked and not checked
- Project field width should change based on whether billable is present or not

**Review by Dimple16 — CHANGES_REQUESTED** [review]
verify all the cases

**Comment by siddharth0-0** [comment]
> Is this tested for the below cases:
> 
> * When projects are assigned  - Tested and able to save the expense
> * When projects are not assigned but it is mandatory - Tested and message is hidden in this case
> * When billable is checked and not checked - Tested and able to save expense
> * Project field width should change based on whether billable is present or not - Tested and looks fine

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.html [hunk 1]
```diff
@@ -255,17 +255,17 @@
...
                 <ng-container
-                  *ngIf="txnFields?.project_id && (isProjectsEnabled$|async) && (((isIndividualProjectsEnabled$|async) && (individualProjectIds$|async)?.length > 0) || !(isIndividualProjectsEnabled$|async))"
+                  *ngIf="(isProjectsEnabled$|async) && (((isIndividualProjectsEnabled$|async) && (individualProjectIds$|async)?.length > 0) || !(isIndividualProjectsEnabled$|async))"
                 >
                   <ng-container *ngIf="projectCategoryIds$|async as projectCategoryIds">
                     <ng-container *ngIf="isProjectVisible$|async">
                       <div
                         class="add-edit-mileage--internal-block add-edit-mileage--project-block"
-                        [ngClass]="{'add-edit-mileage--no-project': fg.controls.project.touched && !fg.controls.project.valid}"
+                        [ngClass]="{'add-edit-mileage--no-project': !fg.controls.project.value}"
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.html:268]: why change this condition?

Comment (siddharth0-0) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.html:268]: In Css  --no-project is set to 'margin-right: 0px !important' 
Earlier when was selecting project as 'None' it was giving a certain margin which was making select dropdown look short 

InkedScreenshot_20211129-115502_Fyle_LI

Comment (Julias0) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.html:268]: but this css class will still be added under certain conditions right? Will it look ok in those cases?

Comment (siddharth0-0) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.html:268]: Yes, --no-project will only be used under when the project value is none.
This is the only time when this --no-project class will be used, In other condition when project value is !Null there will be an checkbox option on right.
image
In this condition the select dropdown will have 'margin-right: 20px;'

**Review by Julias0 — APPROVED** [review]

**Pull request merged**